### ISSUE TITLE
Is there a way to achieve an inset shadow?

### ISSUE THREAD

**Issue by jossmac** [facebook/react-native#2255]
I've tried a myriad of combinations using `shadowOffset` without success. For context this would be used on a `TextInput` but any reasonable number of wrappers would be acceptable.

**Comment by nicklockwood** [comment]
The simplest solution is probably to use a scalable "9-part" image placed inside the TextInput.

**Comment by ide** [comment]
For reference the capInsets prop should help with that: http://facebook.github.io/react-native/docs/image.html#capinsets.

**Comment by nicklockwood** [comment]
Yep. There's an example of capInsets at the bottom of the `<Image>` sample in UIExplorer.

**Comment by jossmac** [comment]
That's interesting, thank you both for getting back to me

**Comment by jeanregisser** [comment]
There's also another trick: using a shadow, border and transparent background and hidding the overflow.

![image](https://cloud.githubusercontent.com/assets/57791/9133868/50694aa6-3cff-11e5-9e93-cb44d0cf778c.png)

https://rnplay.org/apps/OgRtiw

**Comment by Alex-ray** [comment]
fwiw I found this guy 😄 

https://github.com/879479119/react-native-shadow

**Comment by skhameneh-strive** [comment]
@jeanregisser your description is accurate, but please share code snippets over links.  As you can see, some links go down.

**Comment by crystalloo** [comment]
@skhameneh-strive 
I figured a temporary way to make this inset shadow look alright but it is definitely informal.

```
<View style={{
    position: 'absolute',
    top: 0,
    height: 10,
    width: someWidthSameAsParentView,
    marginTop: -10,
    backgroundColor: 'rgb(48, 47, 60)',
    alignSelf: 'center',
    shadowColor: 'rgba(0, 0, 0, 0.7)',
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowRadius: 3,
    shadowOpacity: 1
}} />
```

Add this View inside an outer View with the same width.
This just show one horizontal shadow at top. If you want to show vertical shadows by two sides, add another View with the same style with some changes:

`{height: someHeightSameAsParentView, width: 10, right: -10, top: 10}`

Basically playing around with the absolute position of the View with shadow.
Hope this helps some urgent needs.

**Comment by ashour** [comment]
Just FYI, I hacked inset shadows using [react-native-linear-gradient](https://github.com/react-native-community/react-native-linear-gradient).

<img width="284" alt="screen shot 2018-05-03 at 11 39 19 am" src="https://user-images.githubusercontent.com/330905/39572442-164f18f0-4ec7-11e8-81ea-2a0e2b127707.png">

👆🏽was achieved with:

```
<LinearGradient
        style={styles.container}
        locations={[0, 0.15, 0.15]}
        colors={['rgba(200, 200, 218, 0.25)', 'rgba(200, 200, 218, 0.005)', '#eafcff']}
>
        <TextInput
            {/* ... */}
        />
</LinearGradient>

// ....

const styles = StyleSheet.create({
    container: {
        //...
        backgroundColor: '#eafcff',
    },

//...
```
